index.js

```diff
@@ -1,85 +1,150 @@
-import { RGBot } from "rg-bot";
-import RGCTFUtils, { CTFEvent } from "rg-ctf-utils";
-import { Vec3 } from "vec3";
+const { RGBot } = require('rg-bot');
+const { RGCTFUtils } = require('rg-ctf-utils');
+const {
+  handleAttackFlagCarrier,
+  handleAttackNearbyOpponent,
+  handleBotIdlePosition,
+  handleCollectingFlag,
+  handleLootingItems, handleLowHealth,
+  handlePlacingBlocks, handleScoringFlag
+} = require('./lib/MainLoopFunctions');
+
+const armorManager = require('mineflayer-armor-manager')
+
+const {
+  getUnbreakableBlockIds,
+  nearestTeammates,
+  nameForItem
+} = require('./lib/HelperFunctions')
+
+
+let rgCtfUtils = null
+let unbreakable = null
 
 /**
- * This strategy is the simplest example of how to get started 
- * with the rg-bot and rg-ctf-utils packages. The Bot will get 
- * the flag and then run back to base to score.
- * 
- * Ways to extend this code:
- * TODO: What happens when a bot is completing an action and 
- *       another event happens?
- * TODO: How do we respond to item spawn and drop events?
- * TODO: How do we target and attack enemies?
- * TODO: What different states is my bot in, and how can I organize
- *       its behavior based on these states?
- * rg-bot docs: https://github.com/Regression-Games/RegressionBot/blob/main/docs/api.md
- * rg-ctf-utils docs: https://github.com/Regression-Games/rg-ctf-utils
+ * This capture the flag bot covers most possibilities you could have in a main loop bot.
+ * Macro level strategies and tuning are up to you.
  * @param {RGBot} bot The configurable RGBot
  */
 export function configureBot(bot) {
 
-  // Since most blocks can't be broken on Arctic Algorithm Field,
-  // don't allow digging while pathfinding
-  bot.allowDigWhilePathing(false);
+  // Disable rg-bot debug logging.  You can enable this to see more details about rg-bot api calls
+  bot.setDebug(false)
+
+  // Allow parkour so that our bots pathfinding will jump short walls and optimize their path for sprint jumps.
+  bot.allowParkour(true)
+
+  // We recommend disabling this on as you can't dig the CTF map.  Turning this on can lead pathfinding to get stuck.
+  bot.allowDigWhilePathing(false)
+
+  // Setup the rg-ctf-utils with debug logging
+  rgCtfUtils = new RGCTFUtils(bot)
+  rgCtfUtils.setDebug(true)
+
+  // Load the armor-manager plugin (https://github.com/PrismarineJS/MineflayerArmorManager)
+  bot.mineflayer().loadPlugin(armorManager)
+
+  // Information about the unbreakable block types
+  unbreakable = getUnbreakableBlockIds(bot)
+  console.log(`Unbreakable blocks: ${JSON.stringify(unbreakable)}`)
+
```

Comment:
I think we should still have these specific event listeners so this aligns with the Python and TS examples.

```typescript
    // Listeners for key events.  This bot uses these for logging information for debugging.
    // You may use these for actions, but this main loop bot does not
    bot.on('match_ended', async(matchInfo) => {
        const points = matchInfo?.players.find(player => player.username == bot.username())?.metadata?.score
        const captures = matchInfo?.players.find(player => player.username == bot.username())?.metadata?.flagCaptures
        console.log(`The match has ended - I had ${captures} captures and scored ${points} points`)
        matchInProgress = false
    })

    bot.on('match_started', async(matchInfo) => {
        console.log(`The match has started`)
        matchInProgress = true
    })

    bot.on(CTFEvent.FLAG_OBTAINED, async (collector: string) => {
        console.log(`Flag picked up by ${collector}`)
        if (collector == bot.username()) {
            console.log('I have the flag... yippee !!!')
        }
    })

    bot.on(CTFEvent.FLAG_SCORED, async (teamName: string) => {
        console.log(`Flag scored by ${teamName} team`)
    })

    bot.on(CTFEvent.FLAG_AVAILABLE, async (position: Vec3) => {
        console.log('Flag is available')
    })
```

Reply:
Added these